sorting/stooge_sort.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file
+ * @brief [Stooge sort implementation](https://en.wikipedia.org/wiki/Stooge_sort)
+ * in C++
+ * @details
+ * Stooge sort is a recursive sorting algorithm.
+ * It divides the array into 3 parts and proceeds to:
+ *	- sort first two thirds of the array
+ *	- sort last two thirds of the array
+ *  - sort first two thirds of the array
+ * It's time complexity is O(n^(log3/log1.5)), which is about O(n^2.7),
+ * which makes it to be not the most efficient sorting algorithm
+ * on the street on average. Space complexity is O(1).
+ */
+
+#include <vector> /// for vector
+#include <cassert> /// for assert
+#include <algorithm>  /// for std::is_sorted
+#include <iostream>   /// for IO operations
+
+ /**
+  * @brief The stoogeSort() function is used for sorting the array.
+  * @param L - vector of values (int) to be sorted in in place (ascending order)
+  * @param i - the first index of the array (0)
+  * @param j - the last index of the array (L.size() - 1)
+  * @returns void
+  */
+void stoogeSort(std::vector<int>* L, size_t i, size_t j) {
+	if (i < j) {
+		if ((*L)[i] > (*L)[j]) {
+			std::swap((*L)[i], (*L)[j]);
+		}
+		if (j - i > 1) {
+			size_t third = (j - i + 1) / 3;
+			stoogeSort(L, i, j - third);
+			stoogeSort(L, i + third, j);
+			stoogeSort(L, i, j - third);
+		}
+	}
+	return;
```

Comment:
This is logically correct but could you think of a different way to write this? `@alexpantyukhin` gave the correct suggestion but you've marked it as resolved without considering its impact.

```cpp
void stoogeSort(std::vector<int>* L, size_t i, size_t j) {
 	if (i >= j) {
		return;
	}
	if ((*L)[i] > (*L)[j]) {
 		std::swap((*L)[i], (*L)[j]);
 	}
 	if (j - i > 1) {
 		size_t third = (j - i + 1) / 3;
 		stoogeSort(L, i, j - third);
 		stoogeSort(L, i + third, j);
 		stoogeSort(L, i, j - third);
 	}
}
```

Just adding the guard clause

```cpp
 	if (i >= j) {
		return;
	}
```

makes so much difference in nesting. The use of a guard clause makes reading your code easier for people reviewing your code, and remember that code is read more often than written therefore you should write it in a way to make this easier for readers.